File: Python/tax_calculator.py
import dateutil


def calculate_added_tax(orders):
        tax = 0
        for order in orders:
            if order.date < dateutil.parser.parse("2018-01-01T00:00Z"):
                tax += 10
            else:
                tax += 20

            for product in order.products:
                if product.isEvent:
                    tax += product.price.get_amount_in_currency("USD") * 0.25
                else:
                    tax += product.price.get_amount_in_currency("USD") * 0.175

        return tax

File: TypeScript/src/TaxCalculator.ts
import { Order } from "./Order";
import { Util } from "./Util";

export class TaxCalculator {

    static calculateAddedTax(orders: Order[]): number {
        let tax = 0.0;
        for (const order of orders) {
            if (order.getDate() < Util.fromIsoDate("2018-01-01T00:00Z")) {
                tax += 10;
            } else {
                tax += 20;
            }

            for (const product of order.getProducts()) {
                if (product.isEvent()) {
                    tax += product.getPrice().getAmountInCurrency("USD") * 0.25;
                } else {
                    tax += product.getPrice().getAmountInCurrency("USD") * 0.175;
                }
            }
        }

        return tax;
    }

}
